### ISSUE TITLE
Yubikey hardware not detected with Debian installs or Appimage

### ISSUE THREAD

**Issue by JustSavil** [keepassxreboot/keepassxc#8530]
On version 2.7.1 I tried to close the program and rerun it, even restarting the laptop with the Yubikey plugged in. I also tried uninstalling and reinstalling KeepassXC 2.7.1 and it still doesn't detect my Yubikey 5 NFC. I'm using Linux Mint 20.3. The Yubikey is detected on the Yubikey manager and works for other apps so the problem seems to be isolated to not being detected on KeepassXC. I tried the AppImage and the Debian command line sudo apt-get install keepassxc. I also tried installing using software manager and the keys still arent detected. The libyubikey-udev is present.

**Comment by droidmonkey** [comment]
Have you configured one of the slots to do challenge-response?

**Comment by JustSavil** [comment]
I can double check but yeah I did initially and it used to work on detecting the Yubikey but after the /boot got almost full, I used the auto remove in terminal to clear out space of old kernels, then the Yubikey stopped being detected in Keepassxc

**Comment by droidmonkey** [comment]
You likely need to reinstall several packages including the yubikey udev rules. That is a key detail, KeePassXC did not change in between but your system did. 

We are one of a few apps that actually use the challenge response behavior. Most other uses of the yubikey are done as a "human interface device" which doesn't require raw usb access.

**Comment by JustSavil** [comment]
> You likely need to reinstall several packages including the yubikey udev rules. That is a key detail, KeePassXC did not change in between but your system did.
> 
> We are one of a few apps that actually use the challenge response behavior. Most other uses of the yubikey are done as a "human interface device" which doesn't require raw usb access.

I followed the instructions here: https://support.yubico.com/hc/en-us/articles/360013708900-Using-Your-U2F-YubiKey-with-Linux and I checked the udev rules.

output:
Package: libu2f-udev
Status: install ok installed
Priority: optional
Section: libs
Installed-Size: 26
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Architecture: all
Multi-Arch: foreign
Source: libu2f-host
Version: 1.1.10-1
Depends: udev
Description: Universal 2nd Factor (U2F) common files
 Libu2f is a package for doing Universal 2nd Factor (U2F) host communication
 and has functionality for the Registration and Authentication operations.
 .
 This package provides the udev rules file required to support U2F devices.
Original-Maintainer: Debian Authentication Maintainers <team+auth@tracker.debian.org>


What are the other packages?

**Comment by JustSavil** [comment]
> Have you configured one of the slots to do challenge-response?

Yes I set up the challenge response.

**Comment by droidmonkey** [comment]
Run keepassxc as root and see if you can see your key. If so then your udev rules are not being applied.

**Comment by JustSavil** [comment]
> Run keepassxc as root and see if you can see your key. If so then your udev rules are not being applied.

Yes I can see my key when I run as root. How can I apply the udev rules?

**Comment by JustSavil** [comment]
I copied the text from here:
https://github.com/Yubico/yubikey-personalization/blob/master/70-yubikey.rules

Then I put the file here in /etc/udev/rules.d/ and followed by running sudo udevadm control --reload-rules && sudo udevadm trigger.

Still not showing up. I hope to get it applied permission wise so I don't have to run as root everytime.

**Comment by droidmonkey** [comment]
Do you have KPXC installed as a snap or flatpak? You need to provide raw-usb plug permissions if so. Also if you put keepassxc in a firejail or similar you'll need to either not do that or provide a punch out for the usb access. 

Probably worth a reboot as well to ensure your rules get applied.

**Comment by JustSavil** [comment]
> Do you have KPXC installed as a snap or flatpak? You need to provide raw-usb plug permissions if so. Also if you put keepassxc in a firejail or similar you'll need to either not do that or provide a punch out for the usb access.
> 
> Probably worth a reboot as well to ensure your rules get applied.

Not as snap or flatpak. I downloaded the AppImage and I also tried using _sudo apt install keepassxc_ then restarted. No go.

I then tried sudo apt-get install keepassxc and restarted. No go.

**Comment by JustSavil** [comment]
There have to be something better other than running sudo ./KeePassXC-2.7.1-x86_64.AppImage in terminal

**Comment by JustSavil** [comment]
Is snap the same as software manager in linux mint? I tried this command and restarted
sudo snap connect keepassxc:raw-usb core:raw-usb


still doesn't work :(

**Comment by droidmonkey** [comment]
Recommend removing the yubikey from your database credentials (Database -> Database Security). I also recommend trying to run a live version of your OS to prove it has something to do with your configuration. This capability obviously works for everyone else, myself included.

**Comment by JustSavil** [comment]
> Recommend removing the yubikey from your database credentials (Database -> Database Security). I also recommend trying to run a live version of your OS to prove it has something to do with your configuration. This capability obviously works for everyone else, myself included.

Yep I ran Linux Mint Live and on Tails and the Yubikey showed up without any problem. I don't know what happened to the rules when I just install the updates when prompted on Linux mint. When the boot got full, I would delete the kernels and somehow after that it would stop working. I hope if I upgrade to Linux Mint 21 that it would help. Thanks for all your help.

**Comment by JustSavil** [comment]
OMG I got it to workI I did `sudo nano /etc/udev/rules.d/70-u2f.rules` and found the `# Yubico YubiKey `line. I then deleted:

`KERNEL=="hidraw*", SUBSYSTEM=="hidraw", `(because I read somewhere that deleting that part will possibly get it to work) then pressed CTRL-X to save. I then ran the command in terminal `sudo udevadm control --reload-rules && sudo udevadm trigger` and now KeepassXC works without me needing to use root like I used to. !
